presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotQueryGeneratorContext.java

```diff
@@ -359,7 +360,16 @@ else if (!groupByColumns.isEmpty()) {
         }
 
         List<Integer> indices = getIndicesMappingFromPinotSchemaToPrestoSchema(query, getAssignments());
```

Comment:
@fx19880617 , I am wondering if you are aware of this "indices" thing here ? 

We already had support for the "column reordering" by pinot. Pinot always returns the columns in the order of group-by followed by aggregation. And so we had this indices thing to map the columns returned by pinot into what presto wants. This index map can also drop columns: for example, sometimes on a presto query like "select a from t group by a", we add an additional "count(*)" column that maps to index -1, that is then dropped. 

The way this works is that the query generation process creates the index (during split generation) map and then the page source reads columns in that order. The index map is saved in the split. 

Did you consider somehow reusing this index map? I haven't gone through the full diff as yet, but I would be wary of introducing another permutation map. The first one was hard enough to grok :-P